src/utils/getPreparedTodos.ts

```diff
@@ -0,0 +1,27 @@
+import { Status } from '../types/Status';
+import { Todo } from '../types/Todo';
+
+export const getPreparedTodos = (
+  todoList: Todo[],
+  select: Status,
+  query: string,
+) => {
+  const visibleTodos = [...todoList];
+
+  return visibleTodos
+    .filter(todo =>
+      todo.title.toLowerCase().includes(query.trim().toLowerCase()),
+    )
+    .filter(({ completed }) => {
+      switch (select) {
+        case Status.Active:
+          return !completed;
+
+        case Status.Completed:
+          return completed;
+
+        default:
+          return visibleTodos;
+      }
+    });
+};
```

Comment:
No need to filter in case Status.All or query is empty - it may cause performance issues. Also pay attention, why you return all todos in default case?